lookup.go

```diff
@@ -63,6 +59,42 @@ func lookup(name string, context ...interface{}) (interface{}, bool) {
 	return nil, false
 }
 
+const kLookupTagKey = "mustache"
+
+// lookupStructByName will return the field or the method result with the given name.
+// Returns nil, false if no such field or method.
+func lookupStructByName(reflectValue reflect.Value, name string) (interface{}, bool) {
+	field := reflectValue.FieldByName(name)
+	if field.IsValid() {
+		return field.Interface(), truth(field)
+	}
+	method := reflectValue.MethodByName(name)
+	if method.IsValid() && method.Type().NumIn() == 1 {
+		out := method.Call(nil)[0]
+		return out.Interface(), truth(out)
+	}
+
+	for i := 0; i < reflectValue.NumField(); i++ {
+		tag := reflectValue.Type().Field(i).Tag
+		val, ok := tag.Lookup(kLookupTagKey)
+		if !ok {
+			continue
+		}
+		// NOTE: Is there any option we need to add?
+		// Just treat all tag value as the symbol name for now.
+		if val != name {
+			continue
+		}
```

Comment:
Great question! To start, I think it would be sufficient to lookup the entire tag value as you do now. In the future, we could allow modifying the tag name as an option to `mustache.New()`. 

One use case I can see is reusing the `json` tag. As that tag may have other values following the comma (e.g `omitempty`), we may want to split the value and use the first element.

I'd be more than happy to accept it as is though.